app/controllers/mccloud/WasAnotherPensionSchemeController.scala

```diff
@@ -123,7 +122,12 @@ class WasAnotherPensionSchemeController @Inject()(override val messagesApi: Mess
             },
             value =>
               for {
-                updatedAnswers <- Future.fromTry(userAnswersService.set(WasAnotherPensionSchemePage(chargeType, index), value, mode))
+                updatedAnswers <- Future.fromTry(
+                  request.userAnswers
+                    .removeWithPath(SchemePathHelper.basePath(chargeType, index))
```

Comment:
This is to clean up the data before setting it. This is OK but the normal pattern for this sort of thing is to use the cleanup method on the Page object. I suggest it would be worth getting rid of removewithpath method and using cleanup.